### ISSUE TITLE
File.realpath doesn't follow symlinks

### ISSUE THREAD

**Issue by ConradIrwin** [jruby/jruby#538]
I hoped this would just be a case of copying the logic from rubinius, https://github.com/rubinius/rubinius/blob/master/kernel/common/file19.rb#L98, however that code doesn't raise Errno::ENOENT in the case when the symlink is absent or broken rubinius/rubinius#2165. It might be better to copy the code from 1.8.7's Pathname#realpath which has the behaviour I was expecting.

**Comment by LeeXGreen** [comment]
This might have been fixed at somepoint between 1.7.9 and 1.7.18:

```
leeg-mbp-13:~ lee$ ls -l /tmp/
total 16
-rw-r--r--  1 lee   wheel    0 Feb  8 12:36 file.rb
lrwxr-xr-x  1 lee   wheel    7 Feb  8 12:36 link.rb -> file.rb
lrwxr-xr-x  1 lee   wheel    7 Feb  8 12:36 link2.rb -> link.rb

leeg-mbp-13:~ lee$ rvm use 1.7.18
Using /Users/lee/.rvm/gems/jruby-1.7.18

leeg-mbp-13:~ lee$ irb
jruby-1.7.18 :001 > File.realpath('/tmp/link.rb')
 => "/private/tmp/file.rb"
jruby-1.7.18 :002 > File.realpath('/tmp/link2.rb')
 => "/private/tmp/file.rb"
```

**Comment by limhoff-r7** [comment]
JRuby 1.7.14 fixed the bug.

From: Lee Green
Reply-To: jruby/jruby
Date: Sunday, February 8, 2015 at 1:14 PM
To: jruby/jruby
Cc: Luke Imhoff
Subject: Re: [jruby] File.realpath doesn't follow symlinks (#538)


This might have been fixed at somepoint between 1.7.9 and 1.7.18:

leeg-mbp-13:~ lee$ ls -l /tmp/
total 16
-rw-r--r--  1 lee   wheel    0 Feb  8 12:36 file.rb
lrwxr-xr-x  1 lee   wheel    7 Feb  8 12:36 link.rb -> file.rb
lrwxr-xr-x  1 lee   wheel    7 Feb  8 12:36 link2.rb -> link.rb

leeg-mbp-13:~ lee$ rvm use 1.7.18
Using /Users/lee/.rvm/gems/jruby-1.7.18

leeg-mbp-13:~ lee$ irb
jruby-1.7.18 :001 > File.realpath('/tmp/link.rb')
 => "/private/tmp/file.rb"
jruby-1.7.18 :002 > File.realpath('/tmp/link2.rb')
 => "/private/tmp/file.rb"

**Comment by enebo** [comment]
Based on that snippet I am resolving against 1.7.14.   Thanks for the sleuthing!
